src/form/Form.jsx

```diff
@@ -441,20 +441,31 @@ function Form({
     setRawActiveStep(JSON.parse(JSON.stringify(newActiveStep)));
   };
 
-  const commonCallbackProps = {
-    setOptions: updateFieldOptions(steps),
-    setValues: (userVals) => {
-      const values = convertFilesToFilePromises(userVals, fileServarKeys);
-      updateFieldValues(values);
-      client.submitCustom(values);
-    },
-    setProgress: (val) => setUserProgress(val),
-    setStep: (stepKey) => {
-      changeStep(stepKey, activeStep.key, steps, history);
-    },
-    userId: initInfo().userKey,
-    stepName: activeStep?.key ?? ''
+  const getCommonCallbackProps = (newStep) => {
+    if (!newStep) newStep = activeStep;
+    return {
+      setOptions: updateFieldOptions(steps),
+      setValues: (userVals) => {
+        const values = convertFilesToFilePromises(userVals, fileServarKeys);
+        updateFieldValues(values);
+        client.submitCustom(values);
+      },
+      setProgress: (val) => setUserProgress(val),
+      setStep: (stepKey) => {
+        changeStep(stepKey, newStep.key, steps, history);
+      },
+      step: {
+        style: {
+          backgroundColor: newStep?.default_background_color
+        }
+      },
+      userId: initInfo().userKey,
+      stepName: newStep?.key ?? ''
+    };
   };
+
+  const commonCallbackProps = getCommonCallbackProps();
```

Comment:
can we just have every place call `getCommonCallbackProps` to keep the interface the same